python/graphscope/client/session.py

```diff
@@ -610,15 +614,18 @@ def __init__(
         if self._cluster_type == types_pb2.K8S:
             engines = set([item.strip() for item in enabled_engines.split(",")])
             valid_engines = set(
-                "analytical,analytical-java,interactive,learning,gae,gae-java,gie,gle".split(
+                "analytical,analytical-java,interactive,learning,gae,gae-java,gie,gle,vineyard".split(
                     ","
                 )
             )
-            for item in engines:
-                if item not in valid_engines:
-                    raise ValueError(
-                        f"Not a valid engine name: {item}, valid engines are {valid_engines}"
-                    )
+
+        for item in engines:
+            if item not in valid_engines:
+                raise ValueError(
+                    f"Not a valid engine name: {item}, valid engines are {valid_engines}"
+                )
+            if item == "vineyard":
+                self._with_vineyard = True
```

Comment:
You can add something like `enable_vineyard_scheduler=True/False` rather than initialize a variable `_with_vineyard=False` and match the name `"vineyard"` in enabled engines ...

When you see `with_vineyard=False` you would wondering how graphscope works without vineyard

Reply:
I agree. Vineyard container is always included.

Reply:
rename to something like `create_vineyard_cluster_if_not_exists`